packages/client/ui/react-ui/src/hooks/useAuthSignIn.ts

```diff
@@ -0,0 +1,138 @@
+import type { OAuthProvider } from "@/types/auth";
+import type { UseSignInData } from "@farcaster/auth-kit";
+
+export function useAuthSignIn() {
+    return {
+        onEmailSignIn,
+        onConfirmEmailOtp,
+        onFarcasterSignIn,
+        getOAuthUrl,
+    };
+}
+
+async function onEmailSignIn(email: string, options: { baseUrl: string; apiKey: string }) {
+    try {
+        const response = await fetch(`${options.baseUrl}api/sdk/auth/user/initiate-email-authentication`, {
```

Comment:
they _should_ but lets descope from v0

Reply:
I think all new things should be added in the right place from the get go. You'll see that the good thing about using these services is that they take care of most of the boilerplate.

Reply:
There's gonna be overhead implications if we add them to the crossmint auth service class. For instance, in our CrossmintAuthProvider.tsx file, we already initialize the auth service and we will need to pass that down (most likely in the auth form context provider) and i'm not sure if that's the best way to do it. I don't see why we can't revisit this later as we only support react for right now. If we supported vue I'd move it over but it's not a super necessary optimization imo for right now only because we're trying to move fast

Reply:
The auth client sdk is something we need to ship this week either way, so it needs to be done. You could also instantiate multiple `CrossmintAuthService` if you think that's better. But I think it makes sense that it lives in the `CrossmintAuthProvider` and it can then be fetched from that context which is comfortable to do.

Reply:
If we can merge this right away, I'll do this. I'm already refactoring everything else